### ISSUE TITLE
FMS2021.02-alpha1: ESM4/CM4 models crash with FATAL: input domain does not have an io_domain.

### ISSUE THREAD

**Issue by nikizadehgfdl** [NOAA-GFDL/FMS#714]
**Describe the bug**
New developments in FMS2 forces a fix in MOM6. An io domain with layout 1,1 has to be defined before registering restart axes.  Why is this a requirement? Can it be relaxed?

The coupled models CM4 and ESM4 (that use maintenance branches of MOM6) crash with:

```
FATAL from PE  2134: input domain does not have an io_domain.

Image              PC                Routine            Line        Source
fms_esm4.1_sis2_c  00000000057FC816  Unknown               Unknown  Unknown
fms_esm4.1_sis2_c  00000000035448E1  mpp_mod_mp_mpp_er          68  mpp_util_mpi.inc
fms_esm4.1_sis2_c  000000000344D830  fms_io_utils_mod_         188  fms_io_utils.F90
fms_esm4.1_sis2_c  000000000358F6E2  fms_netcdf_domain         368  fms_netcdf_domain_io.F90
fms_esm4.1_sis2_c  0000000003634F52  coupler_types_mod        3149  coupler_types.F90
fms_esm4.1_sis2_c  000000000040CB47  coupler_main_IP_c        1868  coupler_main.F90
fms_esm4.1_sis2_c  0000000000402642  MAIN__                    612  coupler_main.F90
```
**To Reproduce**
CM4 and ESM4 regression tests

**Expected behavior**
Not crash.

**System Environment**
Describe the system environment, include:
ncrc4.intel18

**Additional context**
@uramirez found the following  fix for this in MOM6. That requires patching two separate maintenance branches of MOM6 (om4/v1.0.5 and ESM4/v1.0.3) neither of which is under FMS control. 

```
git diff MOM6/config_src/coupled_driver/ocean_model_MOM.F90
 use mpp_domains_mod, only : mpp_define_domains, mpp_get_compute_domain, mpp_get_data_domain
+use mpp_domains_mod, only : mpp_define_io_domain

+ call mpp_define_io_domain(Ocean_sfc%Domain, (/1, 1/))
   call mpp_get_compute_domain(Ocean_sfc%Domain, isc, iec, jsc, jec)
```

This could be accomplished (in the xmls) by:
```
(cd src/MOM6 && git remote add nnz https://github.com/nikizadehgfdl/MOM6.git && git fetch nnz && git cherry-pick nnz/FMS2021.02_compliance)
```

I am not sure this fix can easily be extended to the newer updates in MOM6 !

**Comment by menzel-gfdl** [comment]
@nikizadehgfdl fms2 I/O uses the I/O domain to determine how many files are created and which ranks communicate with each other.  I was told when I was doing the fms2 I/O development that every domain "should" have an I/O domain associated with it, but in hindsight we have found that this is rarely true in practice.  Apparently mpp I/O and fms I/O had some way of handling this, but I don't like propagating these hidden behaviors forward.  Is the MOM6 team against adding this extra call to add an I/O domain?

**Comment by nikizadehgfdl** [comment]
@menzel-gfdl , no, they'll do it to their current development branch one way or another. This is about patching old MOM6 maintenance branches to run CM4/ESM4 with preserving original answers.  Also for future reference.
